wordpress-mobile/WordPress-Android
# ViewBinding: Replace Synthetic Accessors in Model Layout/Hope Page Picker Screens

Parent #14845 

This PR is part of a larger effort to replace synthetic accessors within WPAndroid and is split into two parts:
- Convert `ModalLayoutPickerFragment` screen.
- Convert `HomePagePickerFragment` screen.

To test:
- `ModalLayoutPickerFragment` screen:
  - Launch the app.
  - Navigate to `My Site` -> `FAB` -> `Site page`.
  - The `ModalLayoutPickerFragment` screen should be launched.
  - Note that the choose layout shows as expected, and every functionality within works as expected as well.
- `HomePagePickerFragment` screen:
  - Launch the app.
  - Navigate to `My Site` -> `Choose site` -> `+` -> `Create WordPress.com site`
  - The `HomePagePickerFragment` screen should be launched.
  - Note that the choose design shows as expected, and every functionality within works as expected as well.

## Regression Notes
1. Potential unintended areas of impact

N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manually tested as described above in the `To test` section.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:
